library/src/app/components/account-details/account-details.component.spec.ts

```diff
@@ -0,0 +1,217 @@
+
+import { AccountDetailsComponent } from '@components';
+import { TranslateLoader, TranslateModule } from '@ngx-translate/core';
+import { HttpLoaderFactory } from '../../modules/library.module';
+import { HttpClient } from '@angular/common/http';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { SharedModule } from '../../../shared/modules/shared.module';
+import { RouterTestingModule } from '@angular/router/testing';
+import { AssetPipe } from '@pipes';
+import { of, throwError } from 'rxjs';
+import {
+  AccountService,
+  AssetService,
+  ConfigService,
+  ErrorService,
+  EventService,
+  RoutingService
+} from '@services';
+import { TestConstants } from '@constants';
+import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { TradeBankModel, TradesService } from '@cybrid/cybrid-api-bank-angular';
+import { PageEvent } from '@angular/material/paginator';
+
+describe('AccountDetailComponent', () => {
+  let component: AccountDetailsComponent;
+  let fixture: ComponentFixture<AccountDetailsComponent>;
+
+  let MockEventService = jasmine.createSpyObj('EventService', [
+    'getEvent',
+    'handleEvent'
+  ]);
+  let MockErrorService = jasmine.createSpyObj('ErrorService', [
+    'getError',
+    'handleError'
+  ]);
+  let MockConfigService = jasmine.createSpyObj('ConfigService', [
+    'setConfig',
+    'getConfig$'
+  ]);
+  let MockQueryParams = of({
+    accountID: TestConstants.ACCOUNT_GUID
+  });
+  let MockAssetService = jasmine.createSpyObj('AssetService', ['getAsset']);
+  let MockAccountService = jasmine.createSpyObj('AccountService', [
+    'getAccountDetails'
+  ]);
+  let MockTradesService = jasmine.createSpyObj('TradesService', ['listTrades']);
+  let MockRoutingService = jasmine.createSpyObj('RoutingService', [
+    'handleRoute'
+  ]);
+  const error$ = throwError(() => {
+    new Error('Error');
+  });
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      declarations: [AccountDetailsComponent, AssetPipe],
+      imports: [
+        BrowserAnimationsModule,
+        HttpClientTestingModule,
+        SharedModule,
+        RouterTestingModule,
+        TranslateModule.forRoot({
+          loader: {
+            provide: TranslateLoader,
+            useFactory: HttpLoaderFactory,
+            deps: [HttpClient]
+          }
+        })
+      ],
+      providers: [
+        { provide: AssetService, useValue: MockAssetService },
+        { provide: EventService, useValue: MockEventService },
+        { provide: ErrorService, useValue: MockErrorService },
+        { provide: ConfigService, useValue: MockConfigService },
+        { provide: AccountService, useValue: MockAccountService },
+        { provide: TradesService, useValue: MockTradesService },
+        { provide: RoutingService, useValue: MockRoutingService },
```

Comment:
There are a lot of services here. 

I'm worried about the number of API calls made to the backend to load/ interact with this component.

Have you profiled it?

Reply:
It does look like a lot but some of these services don't interact with the client api.

When the app bootstraps is makes 1 call to the assets api. After that it's component specific. 

For the details component it's making a call to: accounts, prices, and trades on initialization, and then to all 3 again every refresh interval as well as a singular call to trades on any pagination event.